lib/webserver/WebServer.js

```diff
@@ -461,23 +461,29 @@ const WebServer = function (options) {
     });
 
     this.app.get("/api/zones", async (req, res) => {
-        // Todo: rename areas to zones in config
-        const zones = this.configuration.get("areas");
-
-        res.json(zones.map(zone => ({
-            name: zone[0],
-            coordinates: zone[1]
-        })));
+        const zones = [...this.configuration.getZones().values()];
+        res.json(zones);
     });
 
     this.app.put("/api/zones", async (req, res) => {
         if (req.body && Array.isArray(req.body)) {
             // Todo: Extended validation
-            const zones = req.body.map(zone => [
-                zone.name,
-                zone.coordinates
-            ]);
-            this.configuration.set("areas", zones);
+            const zones = this.configuration.getZones();
+            const found = new Set();
+            req.body.map(zone => {
+                /** @type {Object} */
+                const oldZone = zones.get(zone.id);
+                if (oldZone) {
```

Comment:
This drops newly created zones.

Reply:
Yes, already fixed in recent commit. Thanks!